src/transformers/models/beit/modeling_beit.py

```diff
@@ -718,12 +715,18 @@ class BeitForMaskedImageModeling(BeitPreTrainedModel):
     def __init__(self, config):
         super().__init__(config)
 
-        self.num_labels = config.num_labels
-        self.beit = BeitModel(config, add_pooling_layer=False)
+        self.beit = BeitModel(config, add_pooling_layer=False, use_mask_token=True)
 
-        # Classifier head
-        self.layernorm = nn.LayerNorm(config.hidden_size, eps=config.layer_norm_eps)
-        self.lm_head = nn.Linear(config.hidden_size, config.vocab_size)
+        if config.decoder_type == "beit":
+            self.layernorm = nn.LayerNorm(config.hidden_size, eps=config.layer_norm_eps)
+            self.lm_head = nn.Linear(config.hidden_size, config.vocab_size)
+        elif config.decoder_type == "simmim":
+            self.decoder = nn.Sequential(
+                nn.Conv2d(in_channels=config.hidden_size, out_channels=config.encoder_stride**2 * 3, kernel_size=1),
+                nn.PixelShuffle(config.encoder_stride),
+            )
+        else:
+            raise ValueError(f"Unknown decoder type {config.decoder_type}. Only 'beit' and 'simmim' are supported.")
```

Comment:
This should probably be raised inside the `config`, so the user gets the error message as early as possible.